components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/GraphQLSchemaDefinition.java

```diff
@@ -77,38 +79,31 @@ public String buildSchemaWithScopesAndRoles(API api) {
         StringBuilder operationAuthSchemeMappingBuilder = new StringBuilder();
         StringBuilder operationThrottlingMappingBuilder = new StringBuilder();
 
-        SwaggerParser parser = new SwaggerParser();
         String swaggerDef = api.getSwaggerDefinition();
+        OpenAPI openAPI = null;
+        LinkedHashMap<String, Object> scopeBindings = null;
 
         if (swaggerDef != null) {
-            swagger = parser.parse(swaggerDef);
+            OpenAPIParser parser = new OpenAPIParser();
+            openAPI = parser.readContents(swaggerDef, null, null).getOpenAPI();
         }
 
-        if (swagger != null) {
-            Map<String, Object> vendorExtensions = swagger.getVendorExtensions();
-            if (vendorExtensions != null) {
-                LinkedHashMap swaggerWSO2Security = (LinkedHashMap) swagger.getVendorExtensions()
-                        .get(APIConstants.SWAGGER_X_WSO2_SECURITY);
-                if (swaggerWSO2Security != null) {
-                    LinkedHashMap swaggerObjectAPIM = (LinkedHashMap) swaggerWSO2Security
-                            .get(APIConstants.SWAGGER_OBJECT_NAME_APIM);
-                    if (swaggerObjectAPIM != null) {
-                        @SuppressWarnings("unchecked")
-                        ArrayList<LinkedHashMap> scopes = (ArrayList<LinkedHashMap>) swaggerObjectAPIM
-                                .get(APIConstants.SWAGGER_X_WSO2_SCOPES);
-                        for (LinkedHashMap scope : scopes) {
-                            for (URITemplate template : api.getUriTemplates()) {
-                                String scopeInURITemplate = template.getScope() != null ?
-                                        template.getScope().getName() : null;
-                                if (scopeInURITemplate != null && scopeInURITemplate.
-                                        equals(scope.get(APIConstants.SWAGGER_SCOPE_KEY))) {
-                                    operationScopeMap.put(template.getUriTemplate(), scopeInURITemplate);
-                                    if (!scopeRoleMap.containsKey(scopeInURITemplate)) {
-                                        scopeRoleMap.put(scopeInURITemplate,
-                                                scope.get(APIConstants.SWAGGER_ROLES).toString());
-                                    }
-                                }
-                            }
+        Map<String, Object> extensions = openAPI.getComponents().getSecuritySchemes()
```

Comment:
What happen swaggerDef is null? If so openAP will be null as well which lead to a null pointer.

Reply:
:+1: